Eahw 2589/set up keycloak for branch deploys

claire-guest opened the pull request:
Deploy keycloak for use by branch deployments.

- drone file inside helm folder will only run if the helm files change
- config is stored in a persistent volume, which will be updated by terraform (PR to follow)

claire-guest committed:
First pass helm files

claire-guest committed:
Add drone file

claire-guest committed:
First deploy attempt

claire-guest committed:
remove paste error

claire-guest committed:
remove branch reference

claire-guest committed:
remove incorrect image ref

claire-guest committed:
hardcode image tag

claire-guest committed:
add service yaml

claire-guest committed:
change service port

claire-guest committed:
change image tag

claire-guest committed:
use user 1000

claire-guest committed:
change port to 8080

claire-guest committed:
add keycloak user

claire-guest committed:
add forwarding env var

claire-guest committed:
change true to string

claire-guest committed:
update to trigger on main only

claire-guest committed:
remove cors

claire-guest committed:
test deploy without cors

claire-guest committed:
run deploy on main only

claire-guest committed:
remove unneeded tag value

sewerynmi commented on helm/templates/deployment.yaml:
```diff
@@ -0,0 +1,41 @@
+        runAsGroup: 1000
+      containers:
+        - name: keycloak
+          image: {{ .Values.image.repo }}keycloak:15.0.2
```
Small suggestion. Is there any sense to keep the Keycloak version as a variable in dev-values.yaml ? Just in case of a version upgrade ?

claire-guest replied:
Updated :)

claire-guest committed:
rename drone steps

claire-guest committed:
move keycloak version to values file

LaurieBennett commented on helm/templates/deployment.yaml:
```diff
@@ -0,0 +1,41 @@
+            - name: http
+              containerPort: 8080
+          env:
+            - name: KEYCLOAK_USER
```
Will this user (with this password) need to be created on the keycloak realm? If so, do we need to add the user to the terraform files so it's managed by terraform?

stephenkaye-ho replied:
This is how the initial username and password is created in order to access the deployed container. It's what needs to be used for the initial terraform apply to create the service account for terraform to use in the pipeline later

LaurieBennett replied:
Ah, ok, that makes sense, no worries on this comment then

LaurieBennett reviewed (changes requested):
One small comment

stephenkaye-ho commented on helm/templates/deployment.yaml:
```diff
@@ -0,0 +1,41 @@
+---
```
I think we might need to add a persistent volume for the config/data so that if the pod is recreated we don't loose state

claire-guest replied:
We've added this now :)

claire-guest committed:
add cors to ingress

stephenkaye-ho committed:
Remove annotations

Annotation for backend protocol is incorrect. Keycloak is running on http
CORS is managed by keycloak so configuring on ingress was overriding keycloak's response

claire-guest committed:
move drone file to helm folder

claire-guest committed:
test helm file move

claire-guest committed:
Merge branch 'main' into eahw-2589/set-up-keycloak-for-branch-deploys

Commit:
Create persistent volume

Commit:
Fix indentation

Commit:
Change mountPath

Commit:
Revert "Change mountPath"

This reverts commit 16b34ca6f7949925bf6a1c4d549acfb195cf2fcb.

Commit:
Change mountPath

Commit:
Change mountPath

claire-guest committed:
copy config and change to recreate strategy

claire-guest committed:
remove dot in path

claire-guest committed:
fix typo

Commit:
add chown

Commit:
add keycloakVolume to chown command

Commit:
change persistent volume access mode

Commit:
experiment with mount path

Commit:
remove jboss from mount path

Commit:
comment out volumes to check file structure

Commit:
Remove contents from keycloak volume

Commit:
Change access mode to ReadWriteOnce

Commit:
Delete contents from keycloakVolume

Commit:
Create new volume

Commit:
Rename persistent volume claim

Commit:
Swap mountPath and name

Commit:
Update claim name

Commit:
Remove slash from mount path

Commit:
Indent volume

Commit:
Add slash

Commit:
Change persistent volume claim to ReadWriteOnce

Commit:
Change mountPath

Commit:
Remove init container

Commit:
Update readme

Commit:
Update drone to deploy on main

Commit:
Remove unused line

claire-guest committed:
add clarification to readme

Commit:
Update readme

stephenkaye-ho commented on README.md:
````diff
@@ -77,4 +77,41 @@ from GitHub Actions, specifying the desired new version in the "Keycloak version
+            - cp -r opt/jboss/. keycloakVolume
+```
+
+It will only need to be run once and the code can be removed after the init container has run. The reason for doing this
````
Can we make this go away as an issue by changing the deployment spec
.spec.strategy.type to recreate

Commit:
Create persistent volume with new mount point

Commit:
Remove dot from copy command

Commit:
Create new volume

Commit:
Change to readWriteMany

Commit:
Change pv name again

Commit:
Change init container name

Commit:
Change pv again

Commit:
Change mountpath back to opt/jboss and create volume 4

Commit:
Use pvc and pv five

Commit:
Add fs group

Commit:
Change mount path

Commit:
Add -n flag to init container to prevent cp overwriting files

Commit:
Update readme

Commit:
Rename pv and pvc

Commit:
Rename pv and pvc to be unique

Commit:
Run on main

Commit:
Remove init container and update readme

joannafawl committed:
Merge branch 'main' into eahw-2589/set-up-keycloak-for-branch-deploys
